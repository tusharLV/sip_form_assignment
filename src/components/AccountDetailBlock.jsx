import { string } from 'prop-types';
const AccountDetailBlock = ({ option, value }) => {
  return (
    <div className="flex my-2">
      <div className="text-gray-400 md:w-1/6 w-1/3 md:text-base text-xs">{option}</div>
      <div className="md:text-base text-xs">:{value}</div>
    </div>
  );
};
AccountDetailBlock.propTypes = {
  option: string,
  value: string
};
export default AccountDetailBlock;
