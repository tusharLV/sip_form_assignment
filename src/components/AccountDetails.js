import {SIPForm, SWPForm, FormHeader} from "./index";
import React from 'react'
import styles from "../style";

const AccountDetails = ({standingInstruction=true}) => {
    return(
        <div className="m-4">
            {
                standingInstruction?(
                       <div>
                           <span className="font-bold mb-4">Registration Mode</span>
                           <div className="my-4">
                               <input type="radio" checked/><span className="ml-4"> Standing Instruction</span>
                           </div>
                           <div className="my-4">
                               Standing instructions are a means to instruct your bank to auto pay a set amount at regular intervals for your investments
                           </div>
                       </div>
                ):(<div></div>)
            }
            <div className="font-semibold my-2">
                Select Payment Bank
            </div>
            <div>
                <select
                    value={"test"}
                    className={`${styles.formInput} w-1/5`}
                >
                    <option value="" label="HDFC Bank Limited/Default" />
                </select>
            </div>
            <div className="flex my-2">
                <div className="text-gray-400 w-1/6">
                    Bank name
                </div>
                <div>
                    :HDFC Bank Limited
                </div>
            </div>
            <div className="flex my-2">
                <div className="text-gray-400 w-1/6">
                    Branch Location
                </div>
                <div>
                    :TULSIANI CHAMBERS - NARIMAN PT/MUMBAI
                </div>
            </div>
            <div className="flex my-2">
                <div className="text-gray-400 w-1/6">
                    Account Type
                </div>
                <div>
                    :SB
                </div>
            </div>
            <div className="flex my-2">
                <div className="text-gray-400 w-1/6">
                    Account Number
                </div>
                <div>
                    :000010109878923
                </div>
            </div>
            <div className="flex my-2">
                <div className="text-gray-400 w-1/6">
                    IFSC Code
                </div>
                <div>
                    :HDFC000000001
                </div>
            </div>
            <div className="flex my-2">
                <div className="text-gray-400 w-1/6">
                    Payment Mode
                </div>
                <div>
                    :Standing Instruction
                </div>
            </div>
        </div>
    )
}
export default AccountDetails
