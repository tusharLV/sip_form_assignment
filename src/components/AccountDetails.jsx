import React from 'react';
import styles from '../style';
import { AccountDetailBlock, RadioInput, Select } from './index';
import { bool } from 'prop-types';
const AccountDetails = ({ standingInstruction = true }) => {
  return (
    <div className="m-4">
      {standingInstruction ? (
        <div>
          <span className="font-bold mb-4">Registration Mode</span>
          <RadioInput checked={true} containerStyle={'my-4'} spanText={'Standing Instruction'} />
          <div className="my-4">
            Standing instructions are a means to instruct your bank to auto pay a set amount at
            regular intervals for your investments
          </div>
        </div>
      ) : (
        <div></div>
      )}
      <div className="font-semibold my-2">Select Payment Bank</div>
      <div>
        <Select
          selectStyle={`${styles.formInput} md:w-1/5 w-2/3`}
          options={[
            {
              value: '',
              label: 'HDFC Bank Limited/Default'
            }
          ]}
        />
      </div>
      <AccountDetailBlock option={'Bank name'} value={'HDFC Bank Limited'} />
      <AccountDetailBlock
        option={'Branch Location'}
        value={'TULSIANI CHAMBERS - NARIMAN PT/MUMBAI'}
      />
      <AccountDetailBlock option={'Account Type'} value={'SB'} />
      <AccountDetailBlock option={'Account Number'} value={'000010109878923'} />
      <AccountDetailBlock option={'IFSC Code'} value={'HDFC000000001'} />
      <AccountDetailBlock option={'Payment Mode'} value={'Standing Instruction'} />
    </div>
  );
};
AccountDetails.propTypes = {
  standingInstruction: bool
};
export default AccountDetails;
