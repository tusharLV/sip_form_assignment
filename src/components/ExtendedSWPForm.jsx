import styles from '../style';
import text from '../constants';
import { AccountDetails, InputHelper, RadioInput, SWPStartDate } from './index';
import { string } from 'prop-types';

const ExtendedSWPForm = ({ switchDate }) => {
  return (
    <div>
      <div className={`${styles.flexContainer}`}>
        <div className={`${styles.subFormContainer}`}>
          <hr className={`${styles.line}`} />
          <div className={`${styles.inputContainer}`}>
            <div className="w-1/3">
              <label htmlFor="">{text.SWP_frequency_label}</label>
            </div>
            <RadioInput containerStyle={'w-1/3'} spanText={'Monthly'} value={'monthly'} />
            <RadioInput containerStyle={'w-1/3'} spanText={'Quarterly'} value={'monthly'} />
          </div>
          <hr className={`${styles.line}`} />
        </div>
      </div>
      <div className={`${styles.flexContainer}`}>
        <div className={`${styles.subFormContainer}`}>
          {text.SWP_amount_label}
          <div className={`${styles.inputContainer} py-4`}>
            <RadioInput
              containerStyle={'md:w-1/3 w-1/2'}
              spanText={'Recommended'}
              value={'monthly'}
            />
            <RadioInput containerStyle={'md:w-1/3 w-1/2'} spanText={'Custom'} value={'quarterly'} />
          </div>
          <div className={`${styles.inputContainer} py-4`}>
            <input type="text" className={`${styles.formInput}`} placeholder={`Amount`} />
          </div>
          <hr className={`${styles.line}`} />
        </div>
        <InputHelper text={text.SWP_amount_input_helper} />
      </div>
      <div className={`${styles.flexContainer}`}>
        <SWPStartDate switchDate={switchDate} />
        <InputHelper text={text.SWP_start_date_helper} />
      </div>
      <AccountDetails standingInstruction={false} />
    </div>
  );
};
ExtendedSWPForm.propTypes = {
  switchDate: string
};
export default ExtendedSWPForm;
