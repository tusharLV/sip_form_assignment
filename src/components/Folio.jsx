import React from 'react';
import styles from '../style';
import { Button, InvestingThrough, Select, FolioHeading } from './index';
import text from '../constants/index';
const Folio = () => {
  return (
    <div className="folio-container md:m-8 m-2 box-border border-2 border-black md:p-8 p-2 rounded">
      <FolioHeading />
      <hr className={`${styles.line}`} />
      <div className="flex">
        <Select
          selectStyle={`${styles.formInput} md:w-1/5 w-1/2 md:text-base text-xs`}
          options={[
            {
              value: '',
              label: 'Choose Folio no'
            },
            {
              value: 'BlackSalt',
              label: 'SNP11289078R'
            }
          ]}
        />
        <div className="md:mx-8 mx-2">OR</div>
        <Button
          styleClass="border-amber-400 border-2 rounded text-amber-400 h-8 bg-transparent md:text-base text-xs md:w-1/5 w-1/2 overflow-ellipsis whitespace-nowrap"
          text={text.proceed_with_a_new_folio_button}
        />
      </div>
      <div className="mt-4">{text.investing_through}</div>
      <InvestingThrough />
    </div>
  );
};
export default Folio;
