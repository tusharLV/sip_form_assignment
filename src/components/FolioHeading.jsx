import text from '../constants';

const FolioHeading = () => {
  return (
    <div>
      <div className="text-2xl font-bold flex items-center">
        {text.folio_heading} &nbsp;
        <img
          src="./exclamation-mark-inside-a-circle-svgrepo-com.svg"
          alt=""
          style={{ height: '20px' }}
        />
      </div>
      <div className="my-4">{text.folio_subheading}</div>
    </div>
  );
};
export default FolioHeading;
