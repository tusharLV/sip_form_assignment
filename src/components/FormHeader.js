import React from 'react'
import styles from "../style";

const FormHeader = ({isSIPFormSubmitted}) => {
    return(
        <div className="mx-16 flex">
            <div className={isSIPFormSubmitted? `${styles.inactiveTab}` : `${styles.activeTab}`}>Step 1: SIP</div>
            <div className={isSIPFormSubmitted? `${styles.activeTab}` : `${styles.inactiveTab}`}>Step 2: SWP</div>
        </div>
    )
}
export default FormHeader
