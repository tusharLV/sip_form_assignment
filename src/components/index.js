import Folio from './Folio';
import SIPForm from './SIPForm';
import SWPForm from './SWPForm';
import FormLayout from './FormLayout';
import FormHeader from './FormHeader';
import AccountDetails from './AccountDetails';
import InvestingThrough from './InvestingThrough';
import Scheme from './Scheme';
import InputHelper from './InputHelper';
import SIPDuration from './SIPDuration';
import SubmitButton from './SubmitButton';
import SWPStartDate from './SWPStartDate';
import Button from './primitives/Button';
import Select from './primitives/Select';
import FolioHeading from './FolioHeading';
import AccountDetailBlock from './AccountDetailBlock';
import ExtendedSWPForm from './ExtendedSWPForm';
import RadioInput from './primitives/RadioInput';
export {
  Folio,
  SIPForm,
  SWPForm,
  FormLayout,
  FormHeader,
  AccountDetails,
  InvestingThrough,
  Scheme,
  InputHelper,
  SIPDuration,
  SubmitButton,
  SWPStartDate,
  Button,
  Select,
  FolioHeading,
  AccountDetailBlock,
  ExtendedSWPForm,
  RadioInput
};
