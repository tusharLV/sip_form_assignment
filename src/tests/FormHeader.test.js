import renderer from 'react-test-renderer';
import { FormHeader } from "../components";

it('should check for FormHeader props and classes', ()=>{
  const component = renderer.create(
    <FormHeader />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree.props).toEqual({ className: 'md:mx-16 mx-4 flex' })
  expect(tree.children.length).toEqual(2)
  expect(tree.children[0].props).toEqual({ className: 'w-1/2 rounded md:h-16 h-8 flex items-center justify-center bg-gray-400 text-white mr-1' })
  expect(tree.children[1].props).toEqual({ className: 'w-1/2 rounded md:h-16 h-8 flex items-center justify-center shadow-xl bg-gray-200 ml-1' })
})