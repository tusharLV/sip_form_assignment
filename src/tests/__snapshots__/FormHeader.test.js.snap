// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should check for FormHeader props and classes 1`] = `
<div
  className="md:mx-16 mx-4 flex"
>
  <div
    className="w-1/2 rounded md:h-16 h-8 flex items-center justify-center bg-gray-400 text-white mr-1"
  >
    Step 1: SIP
  </div>
  <div
    className="w-1/2 rounded md:h-16 h-8 flex items-center justify-center shadow-xl bg-gray-200 ml-1"
  >
    Step 2: SWP
  </div>
</div>
`;
